src/Type/MixedType.php

```diff
@@ -462,8 +462,13 @@ function () use ($level): string {
 
 	public function toBoolean(): BooleanType
 	{
-		if ($this->subtractedType !== null && StaticTypeFactory::falsey()->equals($this->subtractedType)) {
```

Comment:
this never worked because `mixed->equals($x)` always returns false given `$x` is non-mixed

see https://github.com/phpstan/phpstan-src/blob/2c8ab048d91378718116bd42a5e84bf68ed20dac/src/Type/MixedType.php#L287-L289